Build is not working for macOS

If you try to build or run cascette via "cargo run" or "cargo build" it will fail with the following error message:

```
Compiling cascette-client-storage v0.2.0 (/Users/***/***/cascette-rs/crates/cascette-client-storage)
error[E0617]: can't pass u16 to variadic function
  --> crates/cascette-client-storage/src/shmem/platform_unix.rs:69:17
   |
69 |                 (S_IRUSR | S_IWUSR) as mode_t,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: cast the value to c_uint
   |
69 |                 (S_IRUSR | S_IWUSR) as mode_t as c_uint,
   |                                               +++++++++

error[E0617]: can't pass u16 to variadic function
    --> crates/cascette-client-storage/src/shmem/legacy.rs:1314:21
     |
1314 |                     (S_IRUSR | S_IWUSR) as mode_t,
     |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
help: cast the value to c_uint
     |
1314 |                     (S_IRUSR | S_IWUSR) as mode_t as c_uint,
     |                                                   +++++++++

For more information about this error, try rustc --explain E0617.
error: could not compile cascette-client-storage (lib) due to 2 previous errors
```
OS: macOS Tahoe 26.3 
Device: Mac Mini M4

--- a/crates/cascette-client-storage/src/shmem/legacy.rs
+++ b/crates/cascette-client-storage/src/shmem/legacy.rs
@@ -1284,7 +1284,7 @@ mod unix {
     use std::slice;
 
     use libc::{MAP_SHARED, O_CREAT, O_RDWR, PROT_READ, PROT_WRITE, S_IRUSR, S_IWUSR};
-    use libc::{c_void, mode_t, off_t, size_t};
+    use libc::{c_uint, c_void, mode_t, off_t, size_t};
     use libc::{close, ftruncate, mmap, munmap, shm_open, shm_unlink};
 
     use crate::{Result, StorageError};
@@ -1311,7 +1311,7 @@ mod unix {
                 shm_open(
                     c_name.as_ptr(),
                     O_CREAT | O_RDWR,
-                    (S_IRUSR | S_IWUSR) as mode_t,
+                    (S_IRUSR | S_IWUSR) as mode_t as c_uint,
                 )
             };
 

--- a/crates/cascette-client-storage/src/shmem/platform_unix.rs
+++ b/crates/cascette-client-storage/src/shmem/platform_unix.rs
@@ -12,7 +12,7 @@ use std::time::{Duration, Instant};
 use std::{ptr, slice};
 
 use libc::{MAP_SHARED, O_CREAT, O_RDWR, PROT_READ, PROT_WRITE, S_IRUSR, S_IWUSR};
-use libc::{c_void, mode_t, off_t, size_t};
+use libc::{c_uint, c_void, mode_t, off_t, size_t};
 use libc::{close, ftruncate, mmap, munmap, shm_open};
 
 use crate::{Result, StorageError};
@@ -66,7 +66,7 @@ impl PlatformShmem {
             shm_open(
                 c_name.as_ptr(),
                 O_CREAT | O_RDWR,
-                (S_IRUSR | S_IWUSR) as mode_t,
+                (S_IRUSR | S_IWUSR) as mode_t as c_uint,
             )
         };
 

